Backend/Routes/users.js

```diff
@@ -414,4 +415,117 @@ router.get('/connect-requests/mentee/:menteeId', async (req, res) => {
 
 });
 
+// get a mentors meeting
+router.get('/meetings/mentor/:mentorId', async (req, res) => {
+  const { mentorId } = req.params;
+
+  try {
+    const meetings = await Meeting.findAll({
+      where: { mentorId }
+    });
+
+    res.json({ meetings });
+  } catch (error) {
+    res.status(500).json({ error: 'Error fetching meetings' });
+  }
+})
+
+// get a mentees meetings
+router.get('/meetings/mentee/:menteeId', async (req, res) => {
+  const { menteeId } = req.params;
+
+  try {
+    const meetings = await Meeting.findAll({
+      where: { menteeId }
+    });
+
+    res.json({ meetings });
+  } catch (error) {
+    res.status(500).json({ error: 'Error fetching meetings' });
+  }
+})
+
+// check for conflicts
+router.post('/meetings/conflict', async (req, res) => {
+  const { mentorId, menteeId, scheduledTime } = req.body;
+  const startTime = new Date(scheduledTime).toISOString();
+  const endTime = new Date(new Date(scheduledTime).getTime() + 30 * 60 * 1000).toISOString(); // 30 minute meeting
+
+  try {
+    // Check if there is a conflict
+    const conflict = await Meeting.findOne({
+      where: {
+        [Op.or]: [
+          {
+            mentorId,
+            [Op.and]: [
+              { scheduledTime: { [Op.lte]: endTime } },
+              { endTime: { [Op.gte]: startTime } }
+            ]
+          },
+          {
+            menteeId,
+            [Op.and]: [
+              { scheduledTime: { [Op.lte]: endTime } },
+              { endTime: { [Op.gte]: startTime } }
+            ]
+          }
+        ]
+      }
+    });
+
+    res.json({ conflict: !!conflict });
+  } catch (error) {
+    res.status(500).json({ error: 'Error checking for conflicts' });
+  }
+});
+
+// create new meeting
+router.post('/meetings', async (req, res) => {
+  const { mentorId, menteeId, scheduledTime, topic } = req.body;
+  const startTime = new Date(scheduledTime).toISOString();
+  const endTime = new Date(new Date(scheduledTime).getTime() + 30 * 60 * 1000).toISOString(); // 30 minute meeting
+
+  try {
+    // Check if there is a conflict
+    const conflict = await Meeting.findOne({
```

Comment:
what about creating a util for the conflict?

Reply:
I plan on making a util file to refactor my code after I complete my technical challenges. Thanks for the heads up as I will keep this in mind to add to it.